Mutual tls support for falcosidekick authentication (#231)
falcosecurity/falcosidekick: README.md

  outputformat: "all" # all (default), text, fields
  minimumpriority: "debug" # minimum priority of event for using this output, order is emergency|alert|critical|error|warning|notice|informational|debug or "" (default)
  # messageformat: "Alert : rule *{{ .Rule }}* triggered by user *{{ index .OutputFields \"user.name\" }}*" # a Go template to format Rocketchat Text above Attachment, displayed in addition to the output from `ROCKETCHAT_OUTPUTFORMAT`, see [Slack Message Formatting](#slack-message-formatting) in the README for details. If empty, no Text is displayed before Attachment.

>   # mutualtls: false # if true, checkcert flag will be ignored (server cert will always be checked)
>   # checkcert: true # check if ssl certificate of the output is valid (default: true)

Reviewer: ```suggestion
```

Author: Is RocketChat output considered an exclusive SaaS output? I thought you could deploy it on a kubernetes cluster behind an ingresstls object. Maybe the mutualtls flag can be removed, but the checkcert flag too?

Reviewer: I've been confused once again by collapse of line, sorry.

alertmanager:
  # hostport: "" # http://{domain or ip}:{port}, if not empty, Alertmanager output is enabled
  # minimumpriority: "" # minimum priority of event for using this output, order is emergency|alert|critical|error|warning|notice|informational|debug or "" (default)

>   # mutualtls: false # if true, checkcert flag will be ignored (server cert will always be checked)
>   # checkcert: true # check if ssl certificate of the output is valid (default: true)

Reviewer: ```suggestion
```

Author: Why remove this flags from AlertManager output?

loki:
  # hostport: "" # http://{domain or ip}:{port}, if not empty, Loki output is enabled
  # minimumpriority: "" # minimum priority of event for using this output, order is emergency|alert|critical|error|warning|notice|informational|debug or "" (default)
  # checkcert: true # check if ssl certificate of the output is valid (default: true)

nats:
  # hostport: "" # nats://{domain or ip}:{port}, if not empty, NATS output is enabled
  # minimumpriority: "" # minimum priority of event for using this output, order is emergency|alert|critical|error|warning|notice|informational|debug or "" (default)
  # mutualtls: false # if true, checkcert flag will be ignored (server cert will always be checked)
  # checkcert: true # check if ssl certificate of the output is valid (default: true)

stan:
  # hostport: "" # nats://{domain or ip}:{port}, if not empty, STAN output is enabled
  # clusterid: "" # Cluster name, if not empty, STAN output is enabled
  # clientid: "" # Client ID, if not empty, STAN output is enabled
  # minimumpriority: "" # minimum priority of event for using this output, order is emergency|alert|critical|error|warning|notice|informational|debug or "" (default)

>   # checkcert: true # check if ssl certificate of the output is valid (default: true)

Reviewer: ```suggestion
  # mutualtls: false # if true, checkcert flag will be ignored (server cert will always be checked)
  # checkcert: true # check if ssl certificate of the output is valid (default: true)
```

  # apikey: "2c771471-e2af-4dc6-bd35-e7f6ff479b64" # Opsgenie API Key, if not empty, Opsgenie output is enabled
  region: "eu" # (us|eu) region of your domain
  # minimumpriority: "" # minimum priority of event for using this output, order is emergency|alert|critical|error|warning|notice|informational|debug or "" (default)

>   # checkcert: true # check if ssl certificate of the output is valid (default: true)

Reviewer: ```suggestion
```

pagerduty:
  routingKey: "" # Pagerduty Routing Key, if not empty, Pagerduty output is enabled
  minimumpriority: "" # minimum priority of event for using this output, order is emergency|alert|critical|error|warning|notice|informational|debug or "" (default)

>   # checkcert: true # check if ssl certificate of the output is valid (default: true)

Reviewer: ```suggestion
```

  (default: false)
- **CUSTOMFIELDS** : a list of comma separated custom fields to add to falco
  events, syntax is "key:value,key:value"

>   **MUTUALTLSFILESPATH**: path which will be used to stored certs and key for mutual tls authentication

Reviewer: ```suggestion
  **MUTUALTLSFILESPATH**: path which will be used to stored certs and key for mutual tls authentication (default: "/etc/certs")
```

  displayed in addition to the output from `SLACK_OUTPUTFORMAT`, see
  [Slack Message Formatting](#slack-message-formatting) in the README for
  details. If empty, no Text is displayed before Attachment.

> - **SLACK_CHECKCERT** : check if ssl certificate of the output is valid (default:

Reviewer: ```suggestion
```

- **TEAMS_MINIMUMPRIORITY** : minimum priority of event for using use this
  output, order is
  `emergency|alert|critical|error|warning|notice|informational|debug or "" (default)`
- **MATTERMOST_CHECKCERT** : check if ssl certificate of the output is valid (default:
  `true`)
- **DATADOG_APIKEY** : Datadog API Key, if not `empty`, Datadog output is
  _enabled_
- **DATADOG_HOST** : Datadog host. Override if you are on the Datadog EU site.
  Defaults to american site with "https://api.datadoghq.com"
- **DATADOG_MINIMUMPRIORITY** : minimum priority of event for using this output,
  order is
  `emergency|alert|critical|error|warning|notice|informational|debug or "" (default)`

> - **DATADOG_CHECKCERT** : check if ssl certificate of the output is valid (default:

Reviewer: ```suggestion
```

  `GOOGLECHAT_OUTPUTFORMAT`, see
  [Slack Message Formatting](#slack-message-formatting) in the README for
  details. If empty, no Text is displayed before sections.

> - **GOOGLECHAT_CHECKCERT** : check if ssl certificate of the output is valid (default:

Reviewer: ```suggestion
```

- **PAGERDUTY_MINIMUMPRIORITY**: minimum priority of event for using this
  output, order is
  `emergency|alert|critical|error|warning|notice|informational|debug or "" (default)`

> - **PAGERDUTY_CHECKCERT** : check if ssl certificate of the output is valid (default:

Reviewer: ```suggestion
```

2019/05/10 14:32:06 [INFO] : Enabled Outputs : Slack Datadog
```

## Mutual TLS ##

> Outputs with mutualtls enabled in their configuration require *client.crt*, *client.key* and *ca.crt* files to be stored in the path configured in **mutualtlsfilespath** global parameter (**important**: files' name must be preserved)

Reviewer: ```suggestion
Outputs with `mutualtls` enabled in their configuration require *client.crt*, *client.key* and *ca.crt* files to be stored in the path configured in **mutualtlsfilespath** global parameter (**important**: file names must be preserved)
```

```bash
docker run -d -p 2801:2801 -e MUTUALTLSFILESPATH=/etc/certs -e ALERTMANAGER_HOSTPORT=https://XXXX -e ALERTMANAGER_MUTUALTLS=true -e INFLUXDB_HOSTPORT=https://XXXX -e INFLUXDB_MUTUALTLS=true -e WEBHOOK_ADDRESS=XXXX -v /localpath/myclientcert.crt:/etc/certs/client.crt -v /localpath/myclientkey.key:/etc/certs/client.key -v /localpath/ca.crt:/etc/certs/ca.crt falcosecurity/falcosidekick
```

> In above example, the same client certificate will be used for both Alertmanager & InfluxDB outputs which have mutualtls flag set to true.

Reviewer: ```suggestion
In above example, the same client certificate will be used for both Alertmanager & InfluxDB outputs which have mutualtls flag set to true.

```
